From a9e1c25a3ae07bc3d107e75250731654e660725c Mon Sep 17 00:00:00 2001
From: Danny Baumann <dannybaumann@web.de>
Date: Wed, 3 Dec 2014 16:38:57 +0100
Subject: [PATCH 09/17] Fix phone number returned by 'Das Telefonbuch' for
 business numbers.

Some obfuscation spans are inserted into the HTML for whatever reason.

Change-Id: Ib69dac5c79b01947d6085bf27b61c0f7a4d73a4c
Signed-off-by: Chet Kener <Cl3Kener@gmail.com>
---
 src/com/android/dialer/lookup/dastelefonbuch/TelefonbuchApi.java | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/com/android/dialer/lookup/dastelefonbuch/TelefonbuchApi.java b/src/com/android/dialer/lookup/dastelefonbuch/TelefonbuchApi.java
index 2d9ed3e..4e619ed 100644
--- a/src/com/android/dialer/lookup/dastelefonbuch/TelefonbuchApi.java
+++ b/src/com/android/dialer/lookup/dastelefonbuch/TelefonbuchApi.java
@@ -72,6 +72,9 @@ public class TelefonbuchApi {
             boolean dotall, boolean removeSpans) {
         String result = LookupUtils.firstRegexResult(output, regex, dotall);
         if (result != null && removeSpans) {
+            // completely remove hidden spans (including contents) ...
+            result = result.replaceAll("<span class=\"hide\".*?\\/span>", "");
+            // ... and remove span wrappers around data content
             result = result.replaceAll("</?span.*?>", "");
         }
         return LookupUtils.fromHtml(result);
-- 
2.5.0

